### PR TITLE
Removed proper_tea, replaced with param

### PR DESCRIPTION
**Pull request by LiamPattinson**
I switched out proper_tea for the much more widely used package [param](https://param.holoviz.org/index.html). It supports much more than what my own library can do, though I personally think my version is more readable and easier to use.

Advantages of param:

- Better error reporting when setter conditions fail
- Has far more features than the ones I've used here
- Likely to be better supported in future

Disadvantages of param:

- Harder to read. I feel like `positive_float(allow_zero=False)` is easier to read than `Number(1.0, bounds=(0, None), inclusive_bounds=(False, False))`.
- Some parameters require a non-optional default value, such as the example above of a float greater than zero. It may not always be obvious what this should be, and you often have to repeat the default values given in your `__init__` function.

### CODE REVIEW

**Comment by LiamPattinson** [comment]
Another disadvantage of param is that it's quite picky with regards to some input types. For example, the NumericTuple and Range parameters must be given as a tuple, while my implementation would quite happily take any iterable.

File: openmc_plasma_source/fuel_types.py [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
+from param import Parameterized, Number
 
 
-class Fuel:
-    mean_energy = pt.positive_float(allow_zero=False)  # mean energy, eV
-    mass_of_reactants = pt.positive(allow_zero=False)  # mass of the reactants, AMU
+class Fuel(Parameterized):
+
+    # mean energy, eV
+    mean_energy = Number(2.45e6, bounds=(0, None), inclusive_bounds=(False, False))
```
Comment (RemDelaporteMathurin) [inline openmc_plasma_source/fuel_types.py:13]: I think the default value here can be `None`

File: openmc_plasma_source/fuel_types.py [hunk 2]
```diff
@@ -4,12 +4,16 @@
...
+
+    # mass of the reactants, AMU
+    mass_of_reactants = Number(4, bounds=(0, None), inclusive_bounds=(False, False))
```
Comment (RemDelaporteMathurin) [inline openmc_plasma_source/fuel_types.py:16]: I think the default value here can be `None`

File: openmc_plasma_source/ring_source.py [hunk 3]
```diff
@@ -21,11 +20,11 @@ class FusionRingSource(openmc.Source):
         fuel_type (str): The fusion fuel mix. Either 'DT' or 'DD'.
     """
 
-    radius = pt.positive_float()
-    angles = pt.numpy.numpy_array(shape=(2,), dtype=float, sort=True)
-    z_placement = pt.floating_point()
-    temperature = pt.positive_float()
-    fuel_type = pt.in_set(fuel_types.keys())
+    radius = Number(3, bounds=(0, None), inclusive_bounds=(False, False))
```
Comment (RemDelaporteMathurin) [inline openmc_plasma_source/ring_source.py:23]: I think the default value here can be `None`

**Review by RemDelaporteMathurin — APPROVED** [review]

**Comment by LiamPattinson** [comment]
Nicely spotted, I'd assumed Param was using `default=None`, so didn't think setting it to None explicitly would change anything. It seems that instead it's using `default=0`, which fails when bounds are chosen that exclude zero.

I'm now failing the tests because it can't find openmc, but I didn't change any of the import lines since my last commit. Any idea what could be causing this?

**Comment by RemDelaporteMathurin** [comment]
Interesting... it uses openmc:latest and I think there was a release yesterday in openmc. Maybe something was messed up in the docker image. What happens if you try a previous tag of the openmc image?

**Comment by RemDelaporteMathurin** [comment]
Silly me. there isn't any openmc docker image. Forget what I said.

@shimwell  any idea?

**Comment by shimwell** [comment]
That sounds like something has changed inside the openmc docker image, I can look into this

**Comment by shimwell** [comment]
I've reproduced the error, this will need fixing at the openmc end

Screenshot from 2022-02-15 15-04-02

**Comment by RemDelaporteMathurin** [comment]
> Perhaps we can target a different docker image for now
> 
> openmc/openmc:v0.12.2

Yep, another reason to use fixed version tags for dependencies....... :(

**Comment by LiamPattinson** [comment]
Apologies, it looks like I've just pushed the exact same changes as those in pull request https://github.com/fusion-energy/openmc-plasma-source/pull/57.

**Comment by RemDelaporteMathurin** [comment]
@LiamPattinson ah sorry you were faster than me. There are some conflicts now but the tests pass :-)

**Comment by shimwell** [comment]
LGTM

**Pull request merged**